mod-clans/src/main/java/gg/solarmc/loader/clans/Clan.java

```diff
@@ -0,0 +1,407 @@
+
+        int res = transaction.getProperty(DSLContext.class)
+                .insertInto(CLANS_CLAN_ALLIANCES)
+                .columns(CLANS_CLAN_ALLIANCES.CLAN_ID,CLANS_CLAN_ALLIANCES.ALLY_ID)
+                .values(this.clanID,receiver.getID())
+                .execute();
+
+        this.alliedClan = receiver.getID();
+        receiver.alliedClan = this.getID();
+
+        return res == 1;
+    }
+
+    /**
+     * Revokes ally. Can be called from either clan. Revokes alliance for both.
+     * @param transaction the tx
+     * @return whether operation was successful or not
+     */
+    public boolean revokeAlly(Transaction transaction) {
+
+        int i = transaction.getProperty(DSLContext.class)
+                .deleteFrom(CLANS_CLAN_ALLIANCES)
+                .where(CLANS_CLAN_ALLIANCES.CLAN_ID.eq(this.clanID))
+                .or(CLANS_CLAN_ALLIANCES.ALLY_ID.eq(this.clanID))
+                .execute();
+
+        this.getAlliedClan(transaction).ifPresent(c -> c.alliedClan = null);
+        this.alliedClan = null;
+
+        return i == 1;
+    }
+
+    /**
+     * Marks clan as an enemy
+     * @param transaction twix bar
+     * @param receiver the clan to mark as an enemy.
+     * @return whether action was successful or not (failure represents that they are already enemies)
+     * @throws IllegalArgumentException if the presented clan is an ally or is the same object (what?)
+     */
+    public boolean addClanAsEnemy(Transaction transaction, Clan receiver) {
+        if (receiver.getID() == this.clanID) throw new IllegalArgumentException("Tried to mark clan enemy as same clan!");
+
+        this.getAlliedClan(transaction).ifPresent(ally -> {
+            if (this.alliedClan.equals(ally.getID())) throw new IllegalArgumentException("Tried to add allied clan as enemy!");
+        });
+
+        int res = transaction.getProperty(DSLContext.class)
+                .insertInto(CLANS_CLAN_ENEMIES)
+                .columns(CLANS_CLAN_ENEMIES.CLAN_ID,CLANS_CLAN_ENEMIES.ENEMY_ID)
+                .values(this.clanID,receiver.getID())
+                .execute();
+
+        return res == 1;
+    }
+
+    /**
+     * Marks clan enemy as no longer an enemy
+     * @param transaction the tx
+     * @param clan clan provided
+     * @throws IllegalArgumentException if passed self or if passed ally
+     */
+    public void removeClanAsEnemy(Transaction transaction, Clan clan) {
+        if (clan.getID() == this.clanID) throw new IllegalArgumentException("Tried to unmark self as enemy??");
+
+        this.getAlliedClan(transaction).ifPresent(ally -> {
+            if (this.alliedClan.equals(ally.getID())) throw new IllegalArgumentException("Tried to unmark allied clan as enemy!");
+        });
+
+        ClansClanEnemiesRecord rec = transaction.getProperty(DSLContext.class)
+                .fetchOne(CLANS_CLAN_ENEMIES,CLANS_CLAN_ENEMIES.CLAN_ID.eq(this.clanID).and(CLANS_CLAN_ENEMIES.ENEMY_ID.eq(clan.getID())));
```

Comment:
```suggestion
                .fetchOne(CLANS_CLAN_ENEMIES,
                CLANS_CLAN_ENEMIES.CLAN_ID.eq(this.clanID).and(CLANS_CLAN_ENEMIES.ENEMY_ID.eq(clan.getID()))
                .or(
                CLANS_CLAN_ENEMIES.CLAN_ID.eq(clan.getID()).and(CLANS_CLAN_ENEMIES.ENEMY_ID.eq(this.clanID))));
```
Remember, since the clan ID may be the clan_id or the enemy_id, you have to check for both.

Reply:
Clans can be hostile with one clan without the other clan being enemies with them. (Ariel approved this although you could probably ask him again)

Reply:
Huh, okay I'll have to keep that in mind then. Assuming A is hostile with clan B but B is not hostile with A, would A be in clan_id and B be in enemy_id?

Reply:
Yep!